src/pymodaq_data/serialize/factory.py

```diff
@@ -19,21 +21,31 @@ def type(cls):
 
     @staticmethod
     @abstractmethod
-    def serialize(obj: any) -> bytes:
+    def serialize(obj: "SerializableBase") -> bytes:
         "implementation of the serialization into bytes"
         ...
 
     @staticmethod
     @abstractmethod
-    def deserialize(bytes_str: bytes) -> Tuple[Any, bytes]:
+    def deserialize(bytes_str: bytes) -> Tuple["SerializableBase", bytes]:
         "implementation of the deserialization from bytes"
         ...
 
 
+# List of all objects serializable via the serializer
+SERIALIZABLE = Union[bytes, str, int, float, complex, list, NDArray, SerializableBase]
+
+Serializable = TypeVar("Serializable", bound=SERIALIZABLE)
+_SerializableClass = TypeVar("_SerializableClass", bound=SerializableBase)
+
+Serializer = Callable[[Serializable], bytes]
+Deserilizer = Callable[[bytes], Tuple[Serializable, bytes]]
```

Comment:
call it Deserializer

Reply:
actually I'm on it so I will accept the PR and modify the name myself

Reply:
Oh. Thanks